pkg/cloudproviders/aws/cloudprovider/cloudprovider.go

```diff
@@ -293,6 +293,9 @@ func getCABundle(ctx context.Context) *string {
 }
 
 func kubeDNSIP(ctx context.Context, clientSet *kubernetes.Clientset) (net.IP, error) {
+	if clientSet == nil {
```

Comment:
What is the need for this check? Why not just allow the panic and crash?

Reply:
It still crashes, only with a better error message.